src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt

```diff
@@ -305,9 +305,8 @@ function create_and_run_classpath_jar() {
   done
   unset IFS
 
-  RAND_ID=$(cat /dev/urandom | head -c 128 | md5func)
   # Create manifest file
-  MANIFEST_FILE="${self}-${RAND_ID}.jar_manifest"
+  MANIFEST_FILE=$(mktemp)
```

Comment:
```suggestion
  MANIFEST_FILE="$(mktemp -d XXXXXXXX.jar_manifest)"
```

Reply:
It's a bit more tolerant to weird `$TMPDIR`s and the resulting temporary file is better named.

Reply:
@meteorcloudy , do you know why this wasn't using `mktemp` in the first place?

Reply:
For the template you probably wanted something like `/tmp/XXXXXXXX.jar_manifest` instead of `XXXXXXXX.jar_manifest`; see @brown's comment below.